First submit

akiwumi opened the pull request:
Please include your Netlify link here.

akiwumi committed:
Add character counter, error handling, animations, and design system matching image

akiwumi committed:
Rebuild app with component structure and add mock API for local development

akiwumi committed:
Add Netlify link to README

JeffieJansson commented on frontend/src/components/ThoughtForm.css:
```diff
@@ -0,0 +1,137 @@
+  background-color: #ff9d9d;
+  box-shadow: 0 0 12px rgba(255, 0, 0, 0.4);
+  transform: translateY(-1px);
+}
```
Really like the button styling!

JeffieJansson commented on src/App.jsx:
```diff
@@ -1,5 +1,105 @@
+
+const MIN_LENGTH = 5
+const MAX_LENGTH = 140
+
```
the constants MIN_LENGTH and MAX_LENGTH are currently duplicated in both App.jsx and ThoughtForm.jsx. A tip would be to define them in only one file and import them everywhere. :)

akiwumi replied:
thanks will look into cleaning it up

akiwumi replied:
done

akiwumi replied:
first removed from from thoughtForm.jsx, but it caused an error. After debugging I remover the code from the app.jsx

JeffieJansson commented on frontend/src/services/mockApi.js:
Very nice, creating a mock API that uses localStorage!

JeffieJansson commented on frontend/src/components/ThoughtCard.css:
```diff
@@ -0,0 +1,82 @@
+    opacity: 1;
+    transform: translateY(0);
+  }
+}
```
Great Feature! The fadeInUp animation on is a really nice touch.

JeffieJansson reviewed (commented):
Realy like the structure of src/components, src/services and src/utils! And also how you keep the css and jsx files right after one another, makes it really easy to follow the code!
In all I think your code looks really good, great work!

Npahlfer commented on frontend/src/App.jsx:
```diff
@@ -1,5 +1,105 @@
+
+  useEffect(() => {
+    loadThoughts()
+  }, [])
```
You are probably getting a linting warning here that `loadThoughts` are not in the dependency array. 
It would have been better to define the loadThoughts function inside the useEffect.

Npahlfer commented:
You have missed one of the requirements to use the thoughts API. I can see that you have tried to set up your own on Heroku, but that one doesnt work if I visit it and it's not in use in this application. 

Try to refrain from using AI to generate your code and take the time to write it yourself, you will gain from it in the long run!
Try again to do this one, this time without AI :)

akiwumi commented:
I could't find the api, so I had to use a different one. I use AI as a coding partner mainly for debugging and repetitive processes etc. 
Thanks for pointing out the API situation. I've found it so it will be replaced 

> You have missed one of the requirements to use the thoughts API. I can see that you have tried to set up your own on Heroku, but that one doesnt work if I visit it and it's not in use in this application.
> 
> Try to refrain from using AI to generate your code and take the time to write it yourself, you will gain from it in the long run! Try again to do this one, this time without AI :)

akiwumi committed:
Switched API and changed the mock api to false

akiwumi commented:
live api implemented

akiwumi committed:
Remove unused constants from ThoughtForm

akiwumi committed:
Refactor: Create shared constants file and update README

- Created src/constants/index.js for MIN_LENGTH and MAX_LENGTH
- Removed duplicate constants from App.jsx and ThoughtForm.jsx
- Updated README with comprehensive project documentation
- Documented recent API endpoint change
- Added project structure, features, and setup instructions

Npahlfer commented:
Structure wise good. Constants, utils and api methods in separate files. You could have abstracted out into more components even. Eg. having a Button component. 
There is an error ("MAX_LENGTH is not defined") in the deployed code on netlify, so I cant view the new deployment, but I saw the app before the most recent deployment and it looked good :).
Good work!

akiwumi committed:
Restructure project: frontend/ and backend/ directories, add login, add connection guide

akiwumi committed:
Remove FRONTEND_BACKEND_CONNECTION.md

akiwumi committed:
Update README: project structure, tech stack, installation for frontend/backend

akiwumi committed:
feat: Add message editing and deletion functionality

- Add edit and delete endpoints to message controller
- Add edit and delete routes for messages
- Add frontend API methods for editing and deleting messages
- Add UI controls for edit and delete actions in ChatPage
- Add socket events for real-time edit/delete updates
- Add proper TypeScript types and error handling
- Include comprehensive chat system with authentication
- Add deployment documentation and API examples

akiwumi committed:
Fix deployed auth loading and backend build output

akiwumi committed:
Trigger Vercel deploy

akiwumi committed:
Relax CORS frontend origin handling

akiwumi committed:
Keep chat composer visible without an active chat

akiwumi committed:
Allow posting messages without selecting a chat
